### DISCUSSION TITLE
Is it possible to use DataCollection with LiveWire?

### DISCUSSION THREAD

**Discussion by linaspasv** [spatie/laravel-data#412] [Q&A]
Lets say I have the following DTO class.

```php

class MenuData extends Data implements Wireable
{
    use WireableData;

    public function __construct(
        public int $id,
        public string $title,
    ) {

    }
}
```

on a LiveWire component I am trying to mount DTO as this:

```php
    public function mount(): void
    {
        ..........

        $menus = $arena->menus()->get();

        $this->menus = MenuData::collection($menus);
    }
```

and I get ``public property [menus] must be of type: [numeric, string, array, null, or boolean]`` as an error.

I thought that implementing ``Wireable interface`` and using ``WireableData trait`` would create a component that would be auto-transformed into a required data type for a LiveWire component.

Am I missing something? Please advice.

**Comment by rubenvanassche** [comment]
The DataCollection does not support Livewire at the moment due to not adding a dependency on livewire. You have basically two options:

1) Create your own DataCollection extending from ours and implementing the LiveWire stuff and then set the collection property ($_collectionClass) for the required data objects
2) Wait until v4 which allows you to store data in arrays and Collections, natively supported by Livewire

**Comment by francoism90** [comment]
@rubenvanassche Could you give an example? I'm having the same issue and we use V3 on Livewire as well.

**Comment by jskorlol** [comment]
If it's not too much trouble, could I know the expected release date for v4?

**Reply by rubenvanassche** [reply]
No estimates at the moment, probably not before october

**Reply by jskorlol** [reply]
I would like to use it in Livewire. Is there any release schedule for v4? Thank you as always.

**Comment by murindev** [comment]
toCollection() - method does its job even without Wireable if using Livewire 2.
$this->menus = MenuData::collection($menus)->toCollection();
It`s returns Enumerable and works correctly.
Livewire 3 use Wireable in Data class (it is imperative) and returns Collection
